client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the runtime context logging tools
+"""
+
+from concurrent import futures
+import unittest
+import pytest
+from typing import Any, Dict
+from verta import runtime
+
+
+def test_thread_safety() -> None:
+    """
+    Validate that the logging context being logged is thread local.
+    100 threads are completed by 5 workers, which each thread verifying
+    that it only holds it own thread-local value for logs.
+    """
+    def log_in_thread(value):
+        runtime.log('test_key', value)
+        assert runtime._get_thread_logs() == {'test_key': value}
+
+    with futures.ThreadPoolExecutor(max_workers=5) as executor:
+        with runtime.context():
+            executor.map(log_in_thread, [{f'thread_{x}_key': f'thread_{x}_val'} for x in range(101)])
```

Comment:
`log_in_thread()` isn't getting called here. `executor.map()` is lazy: Nothing actually gets evaluated until we actually iterate through the return value from `executor.map()`.

This can be verified by putting an `assert False` in `log_in_thread` and seeing whether `test_thread_safety` passes.

Reply:
Ok, now I understand why many of the examples I found were wrapping the `executor.map()` statement in a `list()` or `print()`.  I couldn't understand why, so I stripped that out on one of the final passes through.  I just wrapped it with list and now changing to `assert False` fails as expected.

Reply:
Fixed (wrapped with `list()`)